### PR TITLE
[DO NOT MERGE]: If token is CT_VBRACE_CLOSE, we must take …

### PR DESCRIPTION
**Pull request by guy-maurel**
…the token before to calculate the value for column.
Ref. #2064

### CODE REVIEW

**Comment by antcodd** [comment]
Is there a more generic solution to this problem than special casing in space.c? It's quite possible there could be other cases. For example I report another one in the same bug #2064 with VBRACE_OPEN, and I've seen infinite growing in unsupported cases (like nested struct table alignment).

Why doesn't the original code break with normal ``{`` braces? Whatever was used there should ideally be used here rather than special casing it. Why does VBRACE_CLOSE in indent.c update pc? I think that may be the source of the inconsistency.

The way uncrustify is written, the fallback and complicated mix of option cases seems to be to mess up the alignment through invalid offset calculations rather than leaving the line as is relative to the basic indent. Maybe something like not popping indent token alignment column changes until after a newline, or a another method of offset calculation (such as separating the concepts of line indent from extra column offset from the previous token in the same line)? There seems to be some infrastructure to this kind of effect in indent.c but there are so many special cases it's hard to follow exactly how it actually works.

**Comment by mihaipopescu** [comment]
@antcodd Thank you for your comment about uncrustify's design.

> Is there a more generic solution to this problem than special casing in space.c?

Now without a massive rewrite but I don't have any _silver bullet_ solution to this.

> infinite growing in unsupported cases (like nested struct table alignment).

I don't think this qualifies for _infinite growing_. Even if something like that is possible in theory, it doesn't mean uncrustify has to support it.

In any case, do not hesitate to make a PR with your proposal in a better direction.
